src/QsFmt/App.Tests/Tests.fs

```diff
@@ -129,3 +144,176 @@ let ``Shows file not found error`` path =
+    )
+
+[<Fact>]
+let ``Input directories`` () =
+    let output =
+        "namespace SubExample1 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace SubExample2 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace SubExample3 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+"
+
+    Assert.Equal(
+        {
+            Code = 0
+            Out = output |> standardizeNewLines
+            Error = ""
+        },
+        run [| "format"; "SubExamples1"; "SubExamples2" |] ""
+    )
+
+[<Fact>]
+let ``Input directories with files and stdin`` () =
+    let input =
+        "namespace StandardIn { function Bar() : Int { return 0; } }
+"
+
+    let output =
+        "namespace Example1 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace StandardIn {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace SubExample1 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace SubExample2 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+"
+
+    Assert.Equal(
+        {
+            Code = 0
+            Out = output |> standardizeNewLines
+            Error = ""
+        },
+        run [| "format"; "Example1.qs"; "-"; "SubExamples1" |] (input |> standardizeNewLines)
+    )
+
+[<Fact>]
+let ``Input directories with recursive flag`` () =
+    let input =
+        "namespace StandardIn { function Bar() : Int { return 0; } }
+"
+
+    let output =
+        "namespace Example1 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace StandardIn {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace SubExample1 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace SubExample2 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace SubExample3 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace NestedExample1 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace NestedExample2 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+"
+
+    Assert.Equal(
+        {
+            Code = 0
+            Out = output |> standardizeNewLines
+            Error = ""
+        },
+        run [| "format"; "-r"; "Example1.qs"; "-"; "SubExamples1"; "SubExamples2" |] (input |> standardizeNewLines)
+    )
+
+[<Fact>]
+let ``Process correct files while erroring on incorrect`` () =
+    let input =
+        "namespace StandardIn { function Bar() : Int { return 0; } }
+"
+
+    let output =
+        "namespace Example1 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace StandardIn {
+    function Bar() : Int {
+        return 0;
+    }
+}
+namespace Example2 {
+    function Bar() : Int {
+        return 0;
+    }
+}
+"
+
+    let result = run [| "format"; "Example1.qs"; "-"; "NotFound.qs"; "Example2.qs" |] (input |> standardizeNewLines)
+    Assert.Equal(3, result.Code)
+    Assert.Equal(output |> standardizeNewLines, result.Out)
+    Assert.NotEmpty(result.Error)
```

Comment:
Is it worth asserting that a specific error is thrown here rather than just any error?

Reply:
This would be a 'file not found' error, and those aren't controlled or maintained by our code. The specific wording might even be dependent on the OS you are running. For that reason, trying to verify that a 'file not found' error is thrown would create a dependency on the test being run in an environment we predict. Just didn't seem that worth it to me to go through that trouble.

Reply:
Makes sense!